src/tokenizer.rs

```diff
@@ -29,19 +29,14 @@ where
     Word(String),
 }
 
+#[inline]
 fn is_whitespace(c: char) -> bool {
-    match c {
-        '\n' | '\r' | '\t' | ' ' => true,
-        _ => false,
-    }
+    c == ' ' || c == '\n' || c == '\r' || c == '\t'
 }
 
+#[inline]
 fn is_numberlike(c: char) -> bool {
-    match c {
-        c if c.is_numeric() => true,
-        '.' | '-' | '+' => true,
-        _ => false,
-    }
+    c == '.' || c == '-' || c == '+' || c.is_ascii_digit()
```

Comment:
I think this is the only part relevant for the correctness fix - the rest of the changes are about perf, right?

Reply:
Yes, correct.